common/app/common/Box.scala

```diff
@@ -0,0 +1,34 @@
+package common
+
+import java.util.concurrent.atomic.AtomicReference
+
+import scala.concurrent.Future
+import scala.util.{Try, Success, Failure}
+
+abstract class Box[T] {
+  def get(): T
+  def apply(): T
+
+  def send(t: T): Unit
+  def send(f: T => T): Unit
+
+  def alter(t: T): Future[T]
+  def alter(t: T => T): Future[T]
+}
+
+object Box {
+  def apply[T](initialValue: T): Box[T] = new AtomicRefBox[T](initialValue)
+}
+
+private class AtomicRefBox[T](initialValue: T) extends Box[T] {
+  private val ref: AtomicReference[T] = new AtomicReference[T](initialValue)
```

Comment:
Note the renaming of the initial value. In `box` this is named [t](https://github.com/guardian/box/blob/master/src/main/scala/com/gu/Box.scala#L28) meaning that in the alter and modify methods `t` was always resolving to this initial value.

Reply:
This comment and [the above one](https://github.com/guardian/frontend/pull/24768/files#r830926884) are good comments, would the be useful to future devs coming to this code?

Reply:
Good shout, will add :)